Extensions/Multiplayer/messageManager.ts

```diff
@@ -1905,19 +2082,26 @@ namespace gdjs {
           const averagePing = Math.round(
             sum / playerLastRoundTripTimes.length / 2 // Divide by 2 to get the one way ping.
           );
-          _playersPings[playerNumber] = averagePing;
+          _playersInfo[playerNumber] = {
+            ping: averagePing,
+            playerId: messageData.playerId,
+            username: messageData.username,
+          };
+
+          // If there are new players, let's resend a heartbeat right away so that everyone is aware of them
+          // on approximately the same frame.
+          if (_playerNumbersWhoJustJoined.length) {
+            const connectedPeerIds = gdjs.multiplayerPeerJsHelper.getAllPeers();
+            const { messageName, messageData } = createHeartbeatMessage();
+            sendDataTo(connectedPeerIds, messageName, messageData);
+            lastHeartbeatTimestamp = getTimeNow();
+          }
         });
       });
     };
 
     const getPlayerPing = (playerNumber: number) => {
-      if (playerNumber < 1) {
-        // Player 1 is the host, so we don't need to compute the ping.
-        // Any negative number is invalid.
-        return 0;
-      }
-
-      return _playersPings[playerNumber] || 0;
+      return (_playersInfo[playerNumber] || {}).ping || 0;
```

Comment:
Will this crash if playerNumber is out of bound? Should we protect this or consider it's ok?

Reply:
it shouldn't crash and just return 0 as ping.